### ISSUE TITLE
tmhoauth worked fine with file based sessions, trying to change to database does not work

### ISSUE THREAD

**Issue by crackers8199** [themattharris/tmhOAuth#114]
i'm stumped.  i've been using the library for around 6 months now with file based php sessions...we're looking to start scaling up now so we want to switch to database sessions, and as soon as i set the database to be the session handler i get 401 unauthorized every time.  the second i change it back to using files, it works fine.

what am i doing wrong?

**Comment by stuarthannig** [comment]
I've never used database sessions before, so sorry I can't help with that.

But I am definitely finding it harder to work with this code outside of sessions, as the functions aren't constructed to return values and are writing to the screen. (NOTE: I am using the example code as my base, because of the lack of documentation apart from the examples.)

It seems like sessions were used as a way to make variables public and share them amongst the functions rather than parametering them in. I wish sessions were just used as a form of pseudo-database rather than storing the working code variables in them.

Basically to convert sessions into variables and arrays,  a lot of code rewrite is needed on the examples.

**Comment by themattharris** [comment]
were you able to resolve this?
as you mentioned the $_SESSION object was used to simplify the examples. that being said, there are also non-interactive examples which also show how to set the token and secret for the request user at runtime.

the library is really a request handler. where the variables that go into it come from doesn't matter to it. they can come from a text file, $_SESSION, a database, or some other system. similarly, the output from the library can be written to the screen, disk or database. 

in tmhOAuthExamples i show how you can reconfigure the token and secret without having to recreate the tmhOAuth object. is that closer to what you were looking for: https://github.com/themattharris/tmhOAuthExamples/blob/master/cli/multi_tokens.php
